sql/layers/layers.go

```diff
@@ -64,6 +65,64 @@ func SetHareOutput(db sql.Executor, lid types.LayerID, output types.BlockID) err
 	return nil
 }
 
+func fill(lid types.LayerID, bid types.BlockID, cert *types.Certificate) {
+	for _, msg := range cert.Signatures {
+		msg.LayerID = lid
+		msg.BlockID = bid
+	}
+}
+
+func prune(cert *types.Certificate) {
```

Comment:
not sure what was the idea but it doesn't prune stuff, it just zeroes it out

Reply:
yikes. changed to use new struct so the field is nil slice